src/enclave/enclave_init.c

```diff
@@ -8,205 +8,202 @@
+    estate->disk_state = oe_calloc(n, sizeof(sgxlkl_enclave_disk_state_t));
+    estate->num_disk_state = n;
 
-    envp = sgxlkl_config->argv + sgxlkl_config->argc + 1;
-    while (envp[envc] != NULL)
+    // root disk index
+    estate->disk_state[0].host_disk_index = 0;
+
+    for (int i = 0; i < cfg->num_mounts; i++)
     {
-        total_size += oe_strlen(envp[envc]) + 1;
-        envc++;
-    }
+        const sgxlkl_enclave_mount_config_t* cfg_disk = &cfg->mounts[i];
 
-    char* buf = oe_malloc((total_size + 1) * sizeof(char*));
-    app_config->argv =
-        oe_malloc((sgxlkl_config->argc + envc + 2) * sizeof(char*));
-    size_t remaining = total_size + 1;
+        if (strcmp(cfg_disk->destination, "/") == 0)
+            sgxlkl_fail("Error: root disk should not be in 'mounts'.\n");
 
-    char* p = buf;
-    for (i = 0; i < app_config->argc; i++)
-    {
-        app_config->argv[i] = p;
-        p += oe_snprintf(p, remaining, "%s", sgxlkl_config->argv[i]) + 1;
-        remaining -= p - app_config->argv[i];
+        bool found = false;
+        for (int j = 0; j < shm->num_virtio_blk_dev && !found; j++)
+        {
+            if (strcmp(cfg_disk->destination, shm->virtio_blk_dev_names[j]) ==
+                0)
+            {
+                estate->disk_state[i + 1].host_disk_index = j;
+                found = true;
+            }
+        }
+        if (!found)
+            sgxlkl_fail(
+                "Disk image for mount point '%s' has not been provided by "
+                "host.\n",
+                cfg_disk->destination);
     }
-    app_config->argv[i] = NULL;
 
-    app_config->envp = &app_config->argv[i + 1];
-    for (i = 0; i < envc; i++)
-    {
-        app_config->envp[i] = p;
-        p += oe_snprintf(p, remaining, "%s", envp[i]) + 1;
-        remaining -= p - app_config->envp[i];
-    }
-    app_config->envp[i] = NULL;
+    lkl_mount_disks(&cfg->root, cfg->mounts, cfg->num_mounts, cfg->cwd);
+}
 
-    app_config->cwd = oe_strdup(sgxlkl_config->cwd);
+// In internal OE header openenclave/internal/sgx/eeid_plugin.h
+#define OE_FORMAT_UUID_SGX_EEID_ECDSA_P256                                \
+    {                                                                     \
+        0x17, 0x04, 0x94, 0xa6, 0xab, 0x23, 0x47, 0x98, 0x8c, 0x38, 0x35, \
+            0x1c, 0xb0, 0xb6, 0xaf, 0x0A                                  \
+    }
 
-    return;
-}
+oe_result_t oe_sgx_eeid_attester_initialize(void);
+oe_result_t oe_sgx_eeid_verifier_initialize(void);
 
-static void enclave_get_app_config(sgxlkl_app_config_t* app_config)
+static void get_attestation_evidence()
 {
-    /* Get the application configuration & HDD param from remote server */
-    if (sgxlkl_enclave->mode == HW_RELEASE_MODE)
+    /* Retrieve remote attestation report to exercise Azure DCAP Client
+     * (currently just for testing) */
+    if (sgxlkl_in_hw_debug_mode() || sgxlkl_in_hw_release_mode())
```

Comment:
Does this always try to attest in HW debug mode? This would again break non-FLC SGX and (some?) SGX2 machines. It would be good if failing attestation in HW debug mode wouldn't be fatal but just generate a warning.

Reply:
It's not run at all (see `if (false)` further down). I've removed this bit completely now, this will be a separate PR.